rfcs/0018-glides-trainsheet-events.md

```diff
@@ -44,36 +44,112 @@ Multiple events can be included in a single Kinesis record, by wrapping them in
 
 ## Value types
 
-### Author
-Each event which is caused by a person will include an `author` key, indicating who performed the given intervention.
+### Car
+Fields:
+- `label` (string | `"cleared"`, optional): car number corresponding to the GTFS-RT `label` field. Or `"cleared"` if the inspector has unassigned the car. If the field is absent, it is not modified.
+- `operator` ([Operator](#Operator) | `"cleared"`, optional): the operator assigned to this car, or the string `"cleared"` if the inspector has unassigned the operator without reassigning another. If the field is absent, it is not modified.
 
-- `emailAddress` (string): the e-mail of the logged-in user.
-- `badgeNumber` (string, optional): the badge number of the logged-in user.
-- `location` (Location, optional): the location the logged-in user is managing.
-```json
-{
-  "emailAddress": "name@example.com",
-  "badgeNumber": "123456",
-  "location": {"gtfsId": "place-lake"}
-}
+An empty object `{}` is valid if nothing about the car has been modified.
 
-or
+The length of `cars` always reflects the known length of the train.
 
-{
-  "emailAddress": "pswartz@mbta.com",
-  "location": {"gtfsId": "place-matt"}
-}
-```
+### DroppedReason
+An object representing the reason that a trip was dropped. The presence of this object implies that the trip was dropped.
+
+Fields:
+- `reason` (string, required): free-text description about why a trip was dropped.
+
+Currently the reason is entered by inspectors in a free-text field. It's published in an object so that if Glides collects more structured data in the future, then extra fields can be added. If more fields are added, then a generated text description would be filled into the `reason` field, and it would not be a breaking change.
+
+### EditorChange
+Fields in the object:
+- `type` (string `"start"`|`"stop"`): Whether the editor started or stopped editing.
+- `location` ([Location](#Location)): the location the editor started or stopped managing.
+- `editor` ([GlidesUser](#GlidesUser)): the affected editor.
+
+### GlidesUser
+Fields:
+- `emailAddress` (string): the e-mail of the user.
+- `badgeNumber` (string, optional): the badge number of the user. Not all Glides users  have a badge number (for example, CTD contractors).
 
 ### Location
 One of:
 - `{"gtfsId": "<GTFS stop ID>"}`, where the GTFS `location_type` is 1 (station).
 - `{"glidesId": "<other>"}`: Other unique identifier for non-revenue locations, internal to Glides.
 
+### Metadata
+Information about how the event was created by Glides. All important outputs from Glides will be elsewhere in the event data, but data here might be useful for consumers who care about how inspectors are using Glides.
+
+Fields:
+- `author` ([GlidesUser](#GlidesUser), optional): The logged-in user whose action triggered the event.
```

Comment:
question: should this be `editor` to match `EditorChange`? Or should `editor` change back to `author`?

Reply:
Neither. The `author` here is always the person who made the change, but the `editor` in `EditorChange` may not be the person who made the change, in the case of taking over editing from someone else. (In the Glides internal audit log, the column name is `target`, but this is because we use the same table for editor changes and trip changes, so I think just `editor` makes sense in `EditorChange`.)